csharp_dotnetcore/8.Suggested-Actions/Suggested_Actions/SuggestedActionsBot.cs

```diff
@@ -33,79 +43,100 @@ public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancel
-                                cancellationToken: cancellationToken);
-                            break;
-                        }
 
-                        case "blue":
-                        {
-                            await turnContext.SendActivityAsync(
-                                "Blue is the best color, I agree.",
-                                cancellationToken: cancellationToken);
-                            break;
-                        }
+                    // Take the input from the user and create the appropriate response.
+                    var responseText = ProcessInput(text);
 
-                        default:
-                        {
-                            await turnContext.SendActivityAsync(
-                                $"Please select a color from the suggested actions",
-                                cancellationToken: cancellationToken);
-                            break;
-                        }
-                    }
+                    // Respond to the user.
+                    await turnContext.SendActivityAsync(responseText, cancellationToken: cancellationToken);
 
                     await SendSuggestedActionsAsync(turnContext, cancellationToken);
                     break;
                 case ActivityTypes.ConversationUpdate:
-
-                    // Send a welcome message to the user and tell them what actions they need to perform to use this bot
+                    // Send a welcome message to the user and tell them what actions they may perform to use this bot
                     if (turnContext.Activity.MembersAdded.Any())
                     {
-                        {
-                            foreach (var member in turnContext.Activity.MembersAdded)
-                            {
-                                var newUserName = member.Name;
-                                if (member.Id != turnContext.Activity.Recipient.Id)
-                                {
-                                    await turnContext.SendActivityAsync(
-                                        $"Welcome to SuggestedActionsBot {member.Name}." +
-                                        $" This bot will introduce you to suggestedActions." +
-                                        $"  Please answer the question:",
-                                        cancellationToken: cancellationToken);
-                                    await SendSuggestedActionsAsync(turnContext, cancellationToken);
-                                }
-                            }
-                        }
+                        await SendWelcomeMessageAsync(turnContext, cancellationToken);
                     }
 
                     break;
             }
         }
 
         /// <summary>
-        /// Creates and sends an activity with suggested actions to the user.
+        /// On a conversation update activity sent to the bot, the bot will
```

Comment:
Consider something like this: "When a conversation update activity is sent to the bot, the bot should send a welcome messages to any new user(s)."